sqlmesh/core/plan/definition.py

```diff
@@ -409,12 +411,16 @@ def _add_restatements(self, restate_models: t.Iterable[str]) -> None:
 
             snapshots = self.context_diff.snapshots
             downstream = [
-                d for d in downstream if snapshots[d].is_materialized and not snapshots[d].is_seed
+                d
+                for d in downstream
+                if snapshots[d].is_materialized
+                and (not snapshots[d].model.disable_restatement or self.is_dev)
+                and not snapshots[d].is_seed
             ]
 
             if not downstream:
                 raise PlanError(
```

Comment:
so just double checking. if you choose to restate something, and a disabled_restatement is in the graph, it will simply be excluded, but it won't be restated is that correct? in that case the down stream of the disabled restatement will still retrigger, but it won't get the new data from this branch

Reply:
> it will simply be excluded, but it won't be restated is that correct? in that case the down stream of the disabled restatement will still retrigger, but it won't get the new data from this branch

All of these statements are correct

Reply:
> disabled restatement will still retrigger, but it won't get the new data from this branch

those downstream might also depend on other upstream that is being restated in addition to the target disabled model.